public/request/game/update-meta.php

```diff
@@ -31,7 +31,7 @@
     return back()->withErrors(__('legacy.error.permissions'));
 }
 
-if (modifyGameData($user, $gameId, $input['developer'], $input['publisher'], $input['genre'], $input['release'], $input['guide_url'])) {
+if (modifyGameData($user, $gameId, trim($input['developer']), trim($input['publisher']), trim($input['genre']), trim($input['release']), trim($input['guide_url']))) {
     return back()->with('success', __('legacy.success.update'));
 }
```

Comment:
Would making `TrimGameMetadata::trimWhitespace` a `public static` method be sufficient?

Also, the trimming should happen within the function as it could potentially be called from other places (it's not currently as far as I can tell).

Reply:
As suggested above, I made `trimWhitespace` a `public static` method and started using it here. Thanks @Jamiras ! I'm not sure I'm _in love_ with that placement; I'm not sure I'd think to look in an Action for this by default. But then again, I'd just double-tap shift in PhpStorm and type `trim` and see it, so maybe it's good enough!

I have a `trim()` in the `trimWhitespace` function, so I _think_ this addresses your concern.

Reply:
You could always put it in `app\Helpers\Util\string.php`, but we're trying to eliminate everything under `app\Helpers`. Maybe @luchaos can offer a suggestion.

Reply:
It doesn't look like you've addressed this:

> Also, the trimming should happen within the function as it could potentially be called from other places (it's not currently as far as I can tell).

Reply:
> It doesn't look like you've addressed this:
> 
> Also, the trimming should happen within the function as it could potentially be called from other places (it's not currently as far as I can tell).

Hm, I'm sorry, I think I'm misunderstanding what you're saying here. In the function `trimWhitespace`, I have the code wrapped in a `trim()`, such that it should be removing extraneous whitespace at the front and back of the string. Could you explain what I'm missing?

Reply:
I'm suggesting the `trim`s be placed inside `modifyGameData` so anything that calls `modifyGameData` will benefit from the trimming without having to do it themselves.

Reply:
Yep, it's embarrassing I didn't immediately understand what you were suggesting. You made a very good point, thanks for explaining! I've made this change in [this commit](https://github.com/RetroAchievements/RAWeb/pull/2125/commits/4282f0e4263e2c57486700ae85f352fdb09a5dd4).